[sumomomomomo] iP

sumomomomomo opened the pull request:
# JoeDuck
> God blessed them and said to them, “Be fruitful and increase in number; fill the earth and subdue it. Rule over the fish in the sea and the birds in the sky and over every living creature that moves on the ground.” - Genesis 1:28 NIV

- Bullet list
1. Numbered list number 1
2. Numbered list number 2
- [ ] Task list task 1
- [ ] Task list task 2

This **word** is bold
This *word* is italic
This ~~word~~ has a line in it
This [word](https://github.com/nus-cs2103-AY2425S1/ip/pull/364) links to this pull request
This `word` is inlined code
Emoji :smiley:

---
These words are in a fenced code block (with syntax highlighting):
```java
public static void main(String[] args) {
    new JoeDuck().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sumomomomomo committed:
Change name to Joe Duck; Add greeting

sumomomomomo committed:
Add echo user input behaviour

sumomomomomo committed:
Store inputs by user

sumomomomomo committed:
Add mark and unmark functionality

sumomomomomo committed:
Add todo, event, deadlines

sumomomomomo committed:
Add initial test

sumomomomomo committed:
Add minimal exception handling

sumomomomomo committed:
Migrate command checking to regex; add delete command

sumomomomomo committed:
Add file save/load skeleton; refactor to use switch statement

sumomomomomo committed:
Add write functionality

sumomomomomo committed:
Finish save and load functionality

sumomomomomo committed:
Dummy empty commit message

sumomomomomo committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Dummy empty commit message

sumomomomomo committed:
Import time related packages

sumomomomomo committed:
Implement LocalDateTime for Deadline and Event

sumomomomomo committed:
Merge branch 'branch-Level-8'

sumomomomomo committed:
Refactor into Ui, Storage, Parser, TaskList

sumomomomomo committed:
Organize classes into packges

sumomomomomo committed:
Merge branch 'add-gradle-support'

sumomomomomo committed:
Create tests for Storage and Parser

sumomomomomo committed:
Fix previous commit

sumomomomomo committed:
Change archiveBaseName to joeduck

sumomomomomo committed:
Add header comments

sumomomomomo committed:
Format code with IntelliJ

sumomomomomo committed:
Implement "find" command

sumomomomomo committed:
Merge branch 'branch-A-JavaDoc'

sumomomomomo committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/joeduck/exception/JoeDuckException.java
#	src/main/java/joeduck/parser/Parser.java
#	src/main/java/joeduck/storage/Storage.java
#	src/main/java/joeduck/task/Todo.java

sumomomomomo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/joeduck/task/TaskList.java

sumomomomomo committed:
Fix previous commit

sumomomomomo committed:
Add GUI

sumomomomomo committed:
Merge branch 'Level-10'

clarud commented on src/main/java/joeduck/JoeDuck.java:
```diff
@@ -0,0 +1,186 @@
+                    return ui.printResponse("Added Event:\n" + e);
+                }
+                default:
+                    throw new InvalidCommandException("Invalid command!");
```
Shouldn't case statement be in line with switch statement?

clarud commented on src/main/java/joeduck/ui/Ui.java:
```diff
@@ -0,0 +1,43 @@
+ * Handles user input/output.
+ */
+public class Ui {
+    private static final String MOTD = "Welcome to Joe Duck";
```
I like that your use of constants is formatted correctly in SCREAMING_SNAKE_CASE.

clarud replied:
Will the use of MOTD acronym be ambiguous?

clarud reviewed (commented):
Awesome coding standard, just 1 thing maybe to fix.

uniqly commented on src/main/java/joeduck/utils/Utils.java:
```diff
@@ -0,0 +1,36 @@
+     * @param forPrinting Boolean representing if the String is for showing to the user.
+     * @return String representation of the List of Task.
+     */
+    public static String inputsToString(List<Task> list, boolean forPrinting) {
```
Perhaps this method could be changed to better reflect what it does?
```suggestion
    public static String getTasksAsString(List<Task> tasks, boolean forPrinting) {
```
Minor nit: would this be better as a instance method of `TaskList`?

uniqly commented on src/main/java/joeduck/ui/Ui.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    public final Scanner scanner = new Scanner(System.in);
+
+    public String printResponse(String res) {
+        System.out.println(res);
+        return res;
+    }
+
+    public String printError(String msg) {
+        printResponse("Error: " + msg);
+        return "Error: " + msg;
+    }
```
I like that you use `print` as a prefix for these methods, makes it clear what they are supposed to do. 👍

uniqly commented on src/main/java/joeduck/ui/Ui.java:
```diff
@@ -0,0 +1,43 @@
+        return scanner.nextLine();
+    }
+
+    public void onStart() {
+        printResponse(MOTD);
+    }
+
+    public String onExit() {
+        printResponse(EXIT_MESSAGE);
+        return EXIT_MESSAGE;
+    }
```
Should there be a `run` or `execute` prefix appended to these methods names (eg: `runOnStart`) to represent that these actions are to be done on the start and on exit?

uniqly commented on src/main/java/joeduck/storage/Storage.java:
```diff
@@ -0,0 +1,168 @@
+
+        String descE = me.group(1);
+        String startDate = me.group(2);
+        LocalDate d1 = LocalDate.parse(startDate);
+        String startTime = me.group(3);
+        LocalTime t1 = LocalTime.parse(startTime);
+        String endDate = me.group(4);
+        LocalDate d2 = LocalDate.parse(endDate);
+        String endTime = me.group(5);
+        LocalTime t2 = LocalTime.parse(endTime);
```
Similar issue in `JoeDuck.java`.

uniqly commented on src/main/java/joeduck/JoeDuck.java:
```diff
@@ -0,0 +1,186 @@
+
+        String desc = me.group(1);
+        String startDate = me.group(2);
+        LocalDate d1 = LocalDate.parse(startDate);
+        String startTime = me.group(3);
+        LocalTime t1 = LocalTime.parse(startTime);
+        String endDate = me.group(4);
+        LocalDate d2 = LocalDate.parse(endDate);
+        String endTime = me.group(5);
+        LocalTime t2 = LocalTime.parse(endTime);
```
The names `d1`, `t1`, etc are not very descriptive, but since `startDate`, etc are already used, perhaps can consider changing `startDate` → `startDateString` and `d1 `→ `startDate`, etc instead?

uniqly commented on src/main/java/joeduck/task/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " +
+                dueDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_PRINT)) + ")";
+    }
+
+    public String toStringWrite() {
+        return "[D]" + super.toString() + " (by: " +
+                dueDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_WRITE)) + ")";
```
Shouldn't the `+` be on the next line when indicating a line break?

uniqly commented on src/main/java/joeduck/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " +
+                startDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_PRINT)) + " to: " +
+                endDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_PRINT)) + ")";
+    }
+
+    public String toStringWrite() {
+        return "[E]" + super.toString() + " (from: " +
+                startDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_WRITE)) + " to: " +
+                endDate.format(DateTimeFormatter.ofPattern(DATE_TIME_PATTERN_WRITE)) + ")";
```
Similar issue in `DeadlineEvent.java`.

uniqly reviewed (commented):
Overall I could understand what the code is doing without much difficulty. However some variable and methods could benefit from more descriptive names. Great effort! 😸

sumomomomomo committed:
Fix coding standard issues

sumomomomomo committed:
Checkstyle: Fix all style errors

Many coding standards are being broken.

Impetus to change was from gradle complaining that checkstyle found errors.

Updated every single file to be compliant to addressbook-3's checkstyle XML.

Refer to the textbook for more information on checkstyle:
https://se-education.org/guides/tutorials/checkstyle.html

sumomomomomo committed:
Implement assertions in Tasks

There are several classes which represent different types of Task.
In all of them, they should never be created with blank arguments.

Adding assertions to related files improves code quality and reliability.

Let's continue adding assertions to areas which may operate on assumptions.

sumomomomomo committed:
Improve code quality

sumomomomomo committed:
Merge pull request #1 from sumomomomomo/branch-A-Assertions

Implement assertions in Tasks

sumomomomomo committed:
Merge pull request #2 from sumomomomomo/branch-A-CodeQuality

Improve code quality

sumomomomomo committed:
Add gradle.yml for CI

sumomomomomo committed:
Implement C-MassOps for mark/unmark/remove

sumomomomomo committed:
Initial AI-assisted refactoring

sumomomomomo committed:
Fix mass command

sumomomomomo committed:
Fix style errors

sumomomomomo committed:
Merge branch 'branch-A-AiAssisted'

sumomomomomo committed:
Improve storage handling

sumomomomomo committed:
Update README.md

sumomomomomo committed:
Edit README.md

sumomomomomo committed:
Fix EOF newlines

sumomomomomo committed:
Remove magic literals from JoeDuck.java

sumomomomomo committed:
Fix header comments
